src/colorizer/ColorizeCanvas.ts

```diff
@@ -522,22 +530,65 @@ export default class ColorizeCanvas {
     return index >= 0 && index < this.getTotalFrames();
   }
 
+  public setObjectOpacity(percentOpacity: number): void {
+    percentOpacity = Math.max(0, Math.min(100, percentOpacity));
+    this.setUniform("objectOpacity", percentOpacity / 100);
+  }
+
+  public setBackdropSaturation(percentSaturation: number): void {
+    percentSaturation = Math.max(0, Math.min(100, percentSaturation));
+    this.setUniform("backdropSaturation", percentSaturation / 100);
+  }
+
+  public setBackdropOpacity(percentOpacity: number): void {
+    percentOpacity = Math.max(0, Math.min(100, percentOpacity));
+    this.setUniform("backdropOpacity", percentOpacity / 100);
+  }
+
+  public setBackdrop(backdropName: string | null): void {
+    this.backdropName = backdropName;
+    this.setFrame(this.currentFrame, true).then(() => {
+      this.render();
+    });
+  }
+
   /**
    * Sets the current frame of the canvas, loading the new frame data if the
    * frame number changes.
    * @param index Index of the new frame.
+   * @param forceUpdate Force a reload of the frame data, even if the frame
+   * is already loaded.
    */
-  async setFrame(index: number): Promise<void> {
+  async setFrame(index: number, forceUpdate: boolean = false): Promise<void> {
     // Ignore same or bad frame indices
-    if (this.currentFrame === index || !this.isValidFrame(index)) {
+    if ((!forceUpdate && this.currentFrame === index) || !this.isValidFrame(index)) {
       return;
     }
```

Comment:
Add an option to force reload. This is used when switching backdrops, where we need to reload the current frame to get the backdrop image.